### PR TITLE
Introduce JWT Access Token Attributes

### PR DESCRIPTION
**Pull request by mpmadhavig**
### Purpose
$subject

### Related PR
- https://github.com/wso2/identity-apps/pull/6730

### Related Issues
- https://github.com/wso2/product-is/issues/20880

New Applications:
Screenshot 2024-08-22 at 13 38 12

Existing Application (Upgrade path):
Screenshot 2024-08-22 at 13 37 54

App list view:
Screenshot 2024-10-15 at 16 22 04

Banner Details:
Screenshot 2024-10-15 at 16 21 56

### CODE REVIEW

**Comment by mpmadhavig** [comment]
Added changes as suggested by @himash-wso2 https://github.com/wso2/identity-apps/pull/6730#issuecomment-2311561546

Screenshot 2024-10-16 at 09 49 32

**Comment by mpmadhavig** [comment]
Application List
Screenshot 2024-10-16 at 10 38 57

Banner View
Screenshot 2024-10-16 at 10 39 04

File: modules/theme/src/themes/default/elements/label.overrides [hunk 1]
```diff
@@ -131,25 +131,19 @@
 
 .ui.label {
     &.issuer-label {
-        color: #e68e00;
```
Comment (JayaShakthi97) [inline modules/theme/src/themes/default/elements/label.overrides:134]: Are these color changes expected?

File: modules/i18n/src/translations/en-US/portals/applications.ts [hunk 2]
```diff
@@ -1672,6 +1690,12 @@ export const applications: ApplicationsNS = {
                                     instruction: "The introspection responses for application access tokens no longer "
                                         + "return the <1>username</1> attribute."
                                 }
+                            },
+                            version200: {
+                                addAllRequestedClaimsInJWTAccessToken: {
+                                    instruction: "Irrespective of the <1>scopes</1> requested, all the <3>Requested Attributes</3> will "
+                                        + "be included in the JWT Access Token."
+                                }
```
Comment (JayaShakthi97) [inline modules/i18n/src/translations/en-US/portals/applications.ts:1698]: No need to break into multiple lines.
```suggestion
                                    instruction: "Irrespective of the <1>scopes</1> requested, all the <3>Requested Attributes</3> will be included in the JWT Access Token."
                                }
```

**Comment by mpmadhavig** [comment]
### Please note the following:

As discussed with @brionmario,
- Used `semantic-ui` `Label` component
    - Changing this to `oxygen-ui` `Chip` component would differ the new label compared to the existing ones.
    - Therefore decided to stick with the same Label component.
    
- As the color used for `issuer` label is similar to  warning color used in `outdated` label AND outdated label should use the warning color, decided to change the `issuer` label color.

File: modules/theme/src/themes/default/elements/label.overrides (hunk 1 above)
Comment (mpmadhavig) [reply modules/theme/src/themes/default/elements/label.overrides:134]: Please refer to https://github.com/wso2/identity-apps/pull/6993#issuecomment-2415817695

File: modules/i18n/src/translations/en-US/portals/applications.ts (hunk 2 above)
Comment (mpmadhavig) [reply modules/i18n/src/translations/en-US/portals/applications.ts:1698]: Wouldn't it reduce the readability of the sentence?

File: .changeset/rich-laws-change.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+"@wso2is/admin.applications.v1": major
+"@wso2is/i18n": major
+---
```
Comment (pavinduLakshan) [inline .changeset/rich-laws-change.md:4]:
```suggestion
"@wso2is/admin.applications.v1": patch
"@wso2is/i18n": patch
---
```

File: features/admin.applications.v1/components/settings/access-configuration.tsx [hunk 4]
```diff
@@ -377,7 +378,7 @@ export const AccessConfiguration: FunctionComponent<AccessConfigurationPropsInte
                         ".genericError.message")
                 }));
             }).finally(() => {
-                onUpdate(appId);
```
Comment (pavinduLakshan) [inline features/admin.applications.v1/components/settings/access-configuration.tsx:380]: Is this intentional?

File: features/admin.applications.v1/pages/application-edit.tsx [hunk 5]
```diff
@@ -707,6 +723,48 @@ const ApplicationEditPage: FunctionComponent<ApplicationEditPageInterface> = (
...
+                                            <Trans
+                                                i18nKey={
+                                                    t("applications:forms.inboundOIDC.sections"
+                                                    + ".outdatedApplications.fields.versions"
+                                                    + ".version200.addAllRequestedClaimsInJWTAccessToken.instruction")
+                                                }
+                                            >
+                                                Irrespective of the <code>scopes</code> requested, all the <code>
+                                                Requested Attributes</code> will be included in the JWT Access Token.
+                                            </Trans>
```
Comment (pavinduLakshan) [inline features/admin.applications.v1/pages/application-edit.tsx:747]:
```suggestion
                                                Irrespective of the <code>scopes</code> requested, all the <code>
                                                requested attributes</code> will be included in the JWT Access Token.
                                            </Trans>
```

File: modules/i18n/src/translations/en-US/portals/applications.ts [hunk 6]
```diff
@@ -1573,6 +1573,24 @@ export const applications: ApplicationsNS = {
...
+                            hint : "Select the attributes that should be included in the <1>access_token</1>.",
+                            label: "Access Token Attributes",
+                            placeholder: "Search by attribute name",
+                            previousBehavior : "Previously, all attributes marked as <1>requested</1> in the " +
+                                "application&apos;s <3>User Attributes</3> section (referred to as requested " +
+                                "attributes) were automatically included in the access token if requested via scopes.",
+                            feature: "With the latest update, admins can now choose which attributes "
+                                + "to include in the access token.",
+                            howToUse : "As this application is eligible to use the feature, you can now see the all " +
+                                "the <1>requested attributes</1> are listed below. Admins can remove/add " +
```
Comment (pavinduLakshan) [inline modules/i18n/src/translations/en-US/portals/applications.ts:1587]:
```suggestion
                                "<1>requested attributes</1> are listed below. Admins can add/remove " +
```

File: modules/i18n/src/translations/en-US/portals/applications.ts [hunk 7]
```diff
@@ -1573,6 +1573,24 @@ export const applications: ApplicationsNS = {
...
+                                "any attributes from the dropdown which includes of <3> requested attributes</3>. " +
```
Comment (pavinduLakshan) [inline modules/i18n/src/translations/en-US/portals/applications.ts:1588]:
```suggestion
                                "any attributes from the dropdown which includes of <3>requested attributes</3>. " +
```

**Comment by himeshsiriwardana** [comment]
> Added changes as suggested by @himeshsiriwardana [#6730 (comment)](https://github.com/wso2/identity-apps/pull/6730#issuecomment-2311561546)
> 
> Screenshot 2024-10-16 at 09 49 32

Previously, if an access token is authorized for a scope, all attributes belonging to that scope and marked as `requested` in the application’s `User Attributes` section, will be included in the access token.

With the latest update, admins can now selectively include user attributes in the access token.

As this application is eligible for this change, you may see the `Access Token Attributes` section below which contains a list of all the requested attributes. If you are an admin, make changes to this list by adding/removing requested attributes. After saving the changes, any future access tokens that are issued will only contain the attributes listed below.

Note:
Once you make changes to this list and save changes, the previous behavior will be permanently replaced. It means that,
- Access tokens will only contain attributes chosen by the admin.
- Admin-chosen attributes are included in the access token even if it does not have authorization to the relevant OIDC scopes.

**Comment by mpmadhavig** [comment]
> > Added changes as suggested by @himeshsiriwardana [#6730 (comment)](https://github.com/wso2/identity-apps/pull/6730#issuecomment-2311561546)
> > Screenshot 2024-10-16 at 09 49 32
> 
> Previously, if an access token is authorized for a scope, all attributes belonging to that scope and marked as `requested` in the application’s `User Attributes` section, will be included in the access token.
> 
> With the latest update, admins can now selectively include user attributes in the access token.
> 
> As this application is eligible for this change, you may see the `Access Token Attributes` section below which contains a list of all the requested attributes. If you are an admin, make changes to this list by adding/removing requested attributes. After saving the changes, any future access tokens that are issued will only contain the attributes listed below.
> 
> Important: Once you make changes to this list and save changes, the previous behavior will be permanently replaced. It means that,
> 
>     * Access tokens will only contain attributes chosen by the admin.
> 
>     * Admin-chosen attributes are included in the access token even if it does not have authorization to the relevant OIDC scopes.
> 
> 
> "Proceed with caution" should be more prominent.

During an offline discussion with @DMHP & @shashimalcse decided to remove the banner in the access token section as we are adding this info in the outdated banner and the docs space. Therefore 7fe2636539474518b43a1071b48bb5485f9b1e34

**Comment by mpmadhavig** [comment]
Screenshot 2024-10-22 at 15 47 44

Screenshot 2024-10-22 at 15 48 02

Screenshot 2024-10-22 at 15 48 16

File: modules/i18n/src/translations/en-US/portals/applications.ts (hunk 2 above)
Comment (JayaShakthi97) [reply modules/i18n/src/translations/en-US/portals/applications.ts:1686]: In FE i18n file, moving forward we do not line break.
Line breaks will make it difficult to search for a string and right away translate to a new language.
Please check other places as well.

File: .changeset/rich-laws-change.md [hunk 8]
```diff
@@ -0,0 +1,6 @@
...
+"@wso2is/admin.applications.v1": patch
+"@wso2is/i18n": patch
+---
```
Comment (JayaShakthi97) [inline .changeset/rich-laws-change.md:4]: Add theme module as well

Comment (mpmadhavig) [reply .changeset/rich-laws-change.md:5]: https://github.com/wso2/identity-apps/pull/6993/commits/607a8048d6856b2ad76c9f9e657411a99ceea040

**Review by JayaShakthi97 — APPROVED** [review]

**Pull request merged**